pkg/pillar/cmd/vaultmgr/vaultmgr.go

```diff
@@ -1083,22 +1112,45 @@ func handleVaultKeyFromControllerImpl(ctxArg interface{}, key string,
 		//Log the type of key used for unlocking default vault
 		log.Noticef("%s unlocked using key type %s", defaultVault,
 			etpm.CompareLegacyandSealedKey().String())
-
-		//Mark the default vault as unlocked
-		ctx.defaultVaultUnlocked = true
-
-		//publish vault key to Controller
-		if err := publishVaultKey(ctx, types.DefaultVaultName); err != nil {
-			log.Errorf("Failed to publish Vault Key, %v", err)
+	case types.EncryptedVaultKeyNoKey:
+		// We are here if we receive no keys from controller
+		if ctx.defaultVaultUnlocked {
+			return
+		}
+		// If we had no luck in unlock with sealed key, and we receive no keys from the controller
+		// than we cannot unlock the vault. It indicates that we did not send key to the controller
+		// for some reason.
```

Comment:
On the TAC call we talked about keeping some persistent state in EVE-OS to reduce the risk of bugs causing accidental destruction of the vault by tracking whether EVE-OS ever managed to successfully send the encrypted key to the controller. I think it makes sense to add such a check here.

Reply:
Yes, I remember. But it is not so easy as I expect during the call. I made changes to store information about sending of storage keys, but also put comment: https://github.com/lf-edge/eve/pull/2755/files#diff-3a4838bcefabcd59acb90ab4d9ffefad17b9dc01ec6e255c05483daeaf4d6064R383-R385.